app/pages/project/floating-ips/FloatingIpsPage.tsx

```diff
@@ -0,0 +1,249 @@
+    },
+  })
+
+  const makeActions = (floatingIp: FloatingIp): MenuAction[] => {
+    const isAttachedToAnInstance = !!floatingIp.instanceId
+    const attachOrDetachAction = isAttachedToAnInstance
+      ? {
+          label: 'Detach',
+          onActivate: () =>
+            confirmAction({
+              actionType: 'danger',
+              doAction: () =>
+                floatingIpDetach.mutateAsync({
+                  path: { floatingIp: floatingIp.name },
+                  query: { project },
+                }),
+              modalTitle: 'Detach Floating IP',
+              modalContent: (
+                <p>
+                  Are you sure you want to detach floating IP <HL>{floatingIp.name}</HL>{' '}
+                  from instance{' '}
+                  <HL>
+                    {
+                      // instanceId is guaranteed to be non-null here
+                      getInstanceName(floatingIp.instanceId!)
+                    }
+                  </HL>
+                  ? The instance will no longer be reachable at <HL>{floatingIp.ip}</HL>.
+                </p>
+              ),
+              errorTitle: 'Error detaching floating IP',
+            }),
+        }
+      : {
+          label: 'Attach',
+          onActivate() {
+            setFloatingIpToModify(floatingIp)
+            setAttachModalOpen(true)
+          },
+        }
+    return [
+      attachOrDetachAction,
+      {
+        label: 'Delete',
+        disabled: isAttachedToAnInstance
+          ? 'This floating IP must be detached from the instance before it can be deleted'
+          : false,
+        onActivate: confirmDelete({
+          doDelete: () =>
+            deleteFloatingIp.mutateAsync({
+              path: { floatingIp: floatingIp.name },
+              query: { project },
+            }),
+          label: floatingIp.name,
+        }),
+      },
+    ]
+  }
+
+  const { Table, Column } = useQueryTable('floatingIpList', { query: { project } })
+  return (
+    <>
+      <PageHeader>
+        <PageTitle icon={<IpGlobal24Icon />}>Floating IPs</PageTitle>
+      </PageHeader>
+      <TableActions>
+        <Link to={pb.floatingIpNew({ project })} className={buttonStyle({ size: 'sm' })}>
+          New Floating IP
+        </Link>
+      </TableActions>
+      <Table emptyState={<EmptyState />} makeActions={makeActions}>
+        <Column accessor="name" />
+        <Column accessor="description" />
+        <Column accessor="ip" />
+        <Column
+          accessor="instanceId"
+          header="Attached to instance"
+          cell={InstanceLinkCell}
+        />
+      </Table>
+      <Outlet />
+      {attachModalOpen && floatingIpToModify && (
+        <AttachFloatingIpModal
+          floatingIp={floatingIpToModify.name}
+          address={floatingIpToModify.ip}
+          instances={instances.items}
+          project={project}
+          onDismiss={() => setAttachModalOpen(false)}
```

Comment:
The whole `attachModalOpen` state variable is unnecessary as you can use whether `floatingIpToModify` is present to mean the same thing. So here you could just do `setFloatingIpToModify(null)` instead, and get rid of all the other references to `attachModalOpen`. Here's another spot where we do this:

https://github.com/oxidecomputer/console/blob/5d989a703359b6fa6723f59a87ba22224e62da34/app/pages/project/networking/VpcPage/tabs/VpcSubnetsTab.tsx#L25